pkg/cmd/admin/registry/registry.go

```diff
@@ -334,3 +334,33 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 	fmt.Fprintf(out, "Docker registry %q service exists\n", name)
 	return nil
 }
+
+func generateProbeConfigForRegistry(ports []kapi.ContainerPort) *kapi.Probe {
+	healthzPort := defaultHealthzPort
+	if len(ports) > 0 {
+		healthzPort = ports[0].ContainerPort
+	}
+
+	return &kapi.Probe{
+		Handler: kapi.Handler{
+			HTTPGet: &kapi.HTTPGetAction{
+				Path: "/",
+				Port: kutil.IntOrString{
+					IntVal: healthzPort,
+				},
+			},
+		},
+	}
+}
+
+func generateLivenessProbeConfig(ports []kapi.ContainerPort) *kapi.Probe {
+	probe := generateProbeConfigForRegistry(ports)
+	if probe != nil {
+		probe.InitialDelaySeconds = 10
```

Comment:
The presence of this line makes me think that `generateProbeConfigForRegistry` isn't factored correctly. It only does a couple of lines of work, so I think inlining it into `generateLivenessProbeConfig` and `generateReadinessProbeConfig` would be better